nitric/api/documents.py

```diff
@@ -0,0 +1,435 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import List, AsyncIterator, Union, Any, Tuple
+
+from nitric.proto.nitric.document.v1 import (
+    DocumentServiceStub,
+    Collection as CollectionMessage,
+    Key as KeyMessage,
+    Expression as ExpressionMessage,
+    ExpressionValue,
+    Document as DocumentMessage,
+)
+from nitric.utils import new_default_channel, _dict_from_struct, _struct_from_dict
+
+
+@dataclass(frozen=True, order=True)
+class DocumentRef:
+    """A reference to a document in a collection."""
+
+    _documents: Documents
+    parent: CollectionRef
+    id: str
+
+    def collection(self, name: str) -> CollectionRef:
+        """
+        Return a reference to a sub-collection of this document.
+
+        This is currently only supported to one level of depth.
+        e.g. Documents().collection('a').doc('b').collection('c').doc('d') is valid,
+        Documents().collection('a').doc('b').collection('c').doc('d').collection('e') is invalid (1 level too deep).
+        """
+        if self.parent.is_sub_collection():
+            # Collection nesting is currently unsupported, but may be included in a future enhancement.
+            raise Exception("Currently, sub-collections may only be nested 1 deep")
```

Comment:
Based on your feedback for the go SDK I think wrapping this in a new exception type and using a constant would be good. In the go sdk we use `"Maximum nested collection depth %d exceed"`. To carry forward the same error for when we open up more depth.

The current message while informative doesn't feel exceptional. Maybe we should add doc links for this error for further explanation like: `"Maximum nested collection depth %d exceed, see: https://nitric.io./<doc_link_with_anchor>"`.